projects/aca-shared/src/lib/services/aca-mobile-app-switcher.service.ts

```diff
@@ -58,18 +58,22 @@ export class AcaMobileAppSwitcherService {
   showAppNotification(): void {
     const ua: string = navigator.userAgent.toLowerCase();
     const isAndroid: boolean = ua.indexOf('android') > -1;
-    const isIphone: boolean = ua.indexOf('iphone') > -1;
+    const isIOS: boolean = this.checkForIOSDevice();
     const url: string = window.location.href;
     const time: number = new Date().getTime();
     sessionStorage.setItem('sessionTime', time.toString());
-    if (isIphone === true) {
+    if (isIOS === true) {
       this.redirectUrl = this.mobileAppSwitchConfig.isIphone + url;
       this.openInApp(this.redirectUrl);
     } else if (isAndroid === true) {
       this.redirectUrl = this.mobileAppSwitchConfig.isAndroidPart1 + url + this.mobileAppSwitchConfig.isAndroidPart2;
       this.openInApp(this.redirectUrl);
     }
   }
+  checkForIOSDevice(): boolean {
+    const ua: string = navigator.userAgent.toLowerCase();
```

Comment:
You already have `ua` variable in `showAppNotification` so instead of creating new one just pass it to `checkForIOSDevice`

Reply:
Also if this function is not being used anywhere else I'm not sure if you even need a function for this as this is just one line check

Reply:
yess I needed it currently for one place to check the ios device type so I removed the function and checked in single line itself.